packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -109,11 +116,27 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
     private async readFromUserHome(): Promise<WorkspaceData | undefined> {
         const file = this.getUserStoragePath();
         if (await fs.pathExists(file)) {
-            const config = await fs.readJson(file);
+            const rawContent = await fs.readFile(file, 'utf-8');
+            const content = rawContent.trim();
+            if (!content) {
+                return undefined;
+            }
+
+            let config;
+            try {
+                config = JSON.parse(content);
+            } catch (error) {
+                this.messageService.warn(`Parse error in '${file}':\nFile will be ignored...`);
```

Comment:
Another thing I am hesitating about: do we expect that this would happen after a user has messed with the recentworkspace.json file, or? Since this file is not vital we could consider deleting it if it's corrupted. At least this way it would start working again after a new workspace is set. WDYT?

Reply:
AFAIK the file is written over when the user loads workspaces, but I could explicitly remove it when an error happens, or maybe not deleting it would allow the user to fix the syntax error ?

What would you prefer ?

Reply:
I am not sure if it's overwritten ATM, but it will probably only be updated after https://github.com/theia-ide/theia/pull/2243.